packages/manager/apps/pci-block-storage/src/api/hooks/useVolume.ts

```diff
@@ -0,0 +1,62 @@
+import { useQuery, useMutation } from '@tanstack/react-query';
+import {
+  deleteVolume,
+  getVolume,
+  getVolumeSnapshot,
+  TVolume,
+  TVolumeSnapshot,
+} from '@/api/data/volume';
+import queryClient from '@/queryClient';
+
+export const getVolumeQueryKey = (projectId: string, volumeId: string) => [
+  'volume',
+  projectId,
+  volumeId,
+];
+
+export const useVolume = (projectId: string, volumeId: string) =>
+  useQuery({
+    queryKey: getVolumeQueryKey(projectId, volumeId),
+    queryFn: (): Promise<TVolume> => getVolume(projectId, volumeId),
+  });
+
+export const getVolumeSnapshotQueryKey = (projectId: string) => [
+  'volume-snapshot',
+  projectId,
+];
+
+export const useVolumeSnapshot = (projectId: string) =>
+  useQuery({
+    queryKey: ['volume-snapshot', projectId],
+    queryFn: (): Promise<TVolumeSnapshot[]> => getVolumeSnapshot(projectId),
+  });
+
+interface DeleteVolumeProps {
+  projectId: string;
+  volumeId: string;
+  onError: (cause: Error) => void;
+  onSuccess: () => void;
+}
+
+export const useDeleteVolume = ({
+  projectId,
+  volumeId,
+  onError,
+  onSuccess,
+}: DeleteVolumeProps) => {
+  const mutation = useMutation({
+    mutationFn: async () => deleteVolume(projectId, volumeId),
+    onError,
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: getVolumeQueryKey(projectId, volumeId),
+      });
+      // @TODO invalidate list of volumes
```

Comment:
Did you forgot to set up the request invalidation ?
Or is it not feasable at this time ?

Reply:
it is not feasable at this time, we'll do it once the listing page will be implemented in this µapp